### ISSUE TITLE
Broken test for .nodeify() return value

### ISSUE THREAD

**Issue by kevinoid** [petkaantonov/bluebird#1030]
This is a very low priority issue, but I thought it would be worth reporting/fixing:

The [returns undefined when a callback is passed](https://github.com/petkaantonov/bluebird/blob/v3.3.3/test/mocha/nodeify.js#L63-L65) test for `.nodeify` incorrectly asserts that `.nodeify` returns `undefined` when a callback is passed.  While it actually [always returns `this`](https://github.com/petkaantonov/bluebird/blob/v3.3.3/src/nodeify.js#L60) and is [documented to do so](http://bluebirdjs.com/docs/api/ascallback.html).  The reason this test doesn't fail is that mocha does not interpret returning `false` as a failure condition and expects an assertion or promise rejection on failure.

Looking into the issue I found #151, which motivated returning `undefined`, and #168, which reverted that change for backward compatibility prior to the 2.0 release.  I assume the behavior is unlikely to change at this point, so the test can be removed or replaced with the [always returns promise for now](https://github.com/petkaantonov/bluebird/blob/v3.3.3/test/mocha/nodeify.js#L90-L99) one from the `isNodeJS` section below.

**Comment by benjamingr** [comment]
related https://github.com/petkaantonov/bluebird/pull/153 and https://github.com/petkaantonov/bluebird/issues/151

**Comment by benjamingr** [comment]
Since #153 was merged we can close this

**Comment by kevinoid** [comment]
@benjamingr No, the test is still wrong.  The test asserts that `.nodeify` "returns undefined when a callback is passed".  It does not, it returns `this`.  However, the test does not fail because it returns false rather than throwing an assertion failure and Mocha ignores falsey return values.

**Comment by benjamingr** [comment]
Oh, so the _test_ is broken? Fair enough

**Comment by kevinoid** [comment]
Indeed.  Sorry about the wall of text which obscured the issue.
